From b80dd201447a52603514404adac47c443735aa8c Mon Sep 17 00:00:00 2001
From: Condor3397 <freedomcondor@126.com>
Date: Tue, 8 May 2018 10:05:32 +0200
Subject: [PATCH] clkgpiochanged

---
 drivers/clk/clk-gpio.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/clk/clk-gpio.c b/drivers/clk/clk-gpio.c
index 151513c655c3..22e1a39753db 100644
--- a/drivers/clk/clk-gpio.c
+++ b/drivers/clk/clk-gpio.c
@@ -35,7 +35,7 @@ static int clk_gpio_gate_enable(struct clk_hw *hw)
 {
 	struct clk_gpio *clk = to_clk_gpio(hw);
 
-	gpiod_set_value(clk->gpiod, 1);
+	gpiod_set_value_cansleep(clk->gpiod, 1);
 
 	return 0;
 }
@@ -44,14 +44,14 @@ static void clk_gpio_gate_disable(struct clk_hw *hw)
 {
 	struct clk_gpio *clk = to_clk_gpio(hw);
 
-	gpiod_set_value(clk->gpiod, 0);
+	gpiod_set_value_cansleep(clk->gpiod, 0);
 }
 
 static int clk_gpio_gate_is_enabled(struct clk_hw *hw)
 {
 	struct clk_gpio *clk = to_clk_gpio(hw);
 
-	return gpiod_get_value(clk->gpiod);
+	return gpiod_get_value_cansleep(clk->gpiod);
 }
 
 const struct clk_ops clk_gpio_gate_ops = {
-- 
2.14.1

